src/pages/home/index.tsx

```diff
@@ -164,9 +164,14 @@ const Home = () => {
   const navigate = useNavigate();
   const dispatch = useAppDispatch();
 
-  const { history, loading } = useAppSelector(state => state.testingHistory);
+  const { history, loading } = useAppSelector((state) => state.testingHistory);
+  const { impersonate, retainId } = useAppSelector((state) => state.profile);
+
+  useEffect(() => {
+    dispatch(fetchHistory(impersonate ? retainId : null));
```

Comment:
We can apply the same principle like the one I suggested for the `createTestRun` action here